routing/triert/table.go

```diff
@@ -0,0 +1,227 @@
+package triert
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+
+	tkey "github.com/libp2p/go-libp2p-xor/key"
+	"github.com/libp2p/go-libp2p-xor/trie"
+	"github.com/plprobelab/go-kademlia/key"
+	"github.com/plprobelab/go-kademlia/network/address"
+)
+
+// TODO: move to key package and replace ErrInvalidKey
+// ErrKeyWrongLength indicates a key has the wrong length
+var ErrKeyWrongLength = errors.New("key has wrong length")
+
+// TrieRT is a routing table backed by a Xor Trie which offers good scalablity and performance
+// for large networks. All exported methods are safe for concurrent use.
+type TrieRT struct {
+	self key.KadKey
+
+	// keymu is held during keys and keyNodes mutations to serialize changes
+	keymu sync.Mutex
```

Comment:
Technically, if this routing table implementation is used by a single thread scheduler, it wouldn't need a lock at all.

Having a lock makes it thread safe, and usable in multi threads scheduler contexts, but we don't have such schedulers yet.

Reply:
I assume there will be other clients of the routing table that will need to do reads. For example the Kubo CLI expects to be able to dump all the entries in a routing table

Reply:
OK, in this case Kubo CLI wouldn't need to take the DHT's lock for it, and would need to enqueue the routing table read action to the scheduler 👍🏻

Reply:
Removed concurrency support from the TrieRT